### PR TITLE
Automatically reset due date for listed versions of incomplete/disabled by user add-ons

### LINKED ISSUE #20198: Listed versions belonging to invisible or incomplete add-ons should not have (automatic) due dates
When the add-on is invisible (`disabled_by_user=True`) or incomplete (`status=STATUS_NULL`, generally missing metadata) its listed versions should not have a due date. They should gain one as soon as the add-on is no longer in one of these states.

### PR DESCRIPTION
**Pull request by diox**
Fixes #20198

### CODE REVIEW

File: src/olympia/versions/models.py [hunk 1]
```diff
@@ -200,7 +201,7 @@ def should_have_due_date(self, negate=False):
                 | requires_manual_listed_approval_and_is_listed
                 | requires_manual_unlisted_approval_and_is_unlisted
             )
-        )
+        ).using('default')
```
Comment (diox) [inline src/olympia/versions/models.py:204]: Drive-by to decrease the possibility of replication-based race conditions

File: src/olympia/versions/models.py [hunk 2]
```diff
@@ -850,6 +851,7 @@ def reset_due_date(self, due_date=None):
             log.info('Version %r (%s) due_date cleared', self, self.id)
             self.update(due_date=None, _signal=False)
 
+    @use_primary_db
```
Comment (diox) [inline src/olympia/versions/models.py:854]: As above

**Review by eviljeff — APPROVED** [review]
r+wc just a question about if one of the changes is necessary but I guess it doesn't hurt 🤷

File: src/olympia/versions/models.py [hunk 3]
```diff
@@ -182,6 +182,7 @@ def should_have_due_date(self, negate=False):
             Q(addon__reviewerflags__auto_approval_disabled=True)
             | Q(addon__reviewerflags__auto_approval_disabled_until_next_approval=True)
             | Q(addon__promotedaddon__group_id__in=(g.id for g in PRE_REVIEW_GROUPS)),
+            addon__status__in=(amo.VALID_ADDON_STATUSES),
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/versions/models.py:185]: https://github.com/mozilla/addons-server/pull/20240/commits/78ec1f144a0527cb97fd53f34f564ed5e2b151be is the test for that change
